basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/referential/component/CityListPanel.java

```diff
@@ -0,0 +1,128 @@
+package fr.openwide.core.basicapp.web.application.referential.component;
+
+import org.apache.wicket.Component;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.form.AjaxFormComponentUpdatingBehavior;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.form.TextField;
+import org.apache.wicket.markup.html.panel.Fragment;
+import org.apache.wicket.model.ResourceModel;
+import org.odlabs.wiquery.core.events.StateEvent;
+
+import fr.openwide.core.basicapp.core.business.referential.model.City;
+import fr.openwide.core.basicapp.core.business.referential.model.search.CitySort;
+import fr.openwide.core.basicapp.core.util.binding.Bindings;
+import fr.openwide.core.basicapp.web.application.referential.form.AbstractGenericListItemPopup;
+import fr.openwide.core.basicapp.web.application.referential.form.CityPopup;
+import fr.openwide.core.basicapp.web.application.referential.model.CityDataProvider;
+import fr.openwide.core.jpa.more.business.generic.model.EnabledFilter;
+import fr.openwide.core.wicket.more.markup.html.factory.AbstractParameterizedComponentFactory;
+import fr.openwide.core.wicket.more.markup.html.feedback.FeedbackUtils;
+import fr.openwide.core.wicket.more.markup.html.form.EnumDropDownSingleChoice;
+import fr.openwide.core.wicket.more.markup.html.form.LabelPlaceholderBehavior;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.DecoratedCoreDataTablePanel.AddInPlacement;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.DataTableBuilder;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.state.IAddedCoreColumnState;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.state.IDecoratedBuildState;
+
+public class CityListPanel extends AbstractGenericListItemListPanel<City, CitySort, CityDataProvider> {
+	
+	private static final long serialVersionUID = -2165029373966420708L;
+	
+	private CityPopup popup;
+	
+	public CityListPanel(String id) {
+		this(id, new CityDataProvider(EnabledFilter.ENABLED_ONLY));
+	}
+	
+	public CityListPanel(String id, CityDataProvider dataProvider) {
+		super(id, dataProvider, dataProvider.getSortModel());
+		setOutputMarkupId(true);
+		
+		add(
+				popup
+		);
+	}
+	
+	@Override
+	protected City getNewInstance() {
+		return new City();
+	}
+
+	@Override
+	protected void createPopup() {
+		popup = new CityPopup("popup") {
```

Comment:
This method could simply return the popup. Then, the responsibility for storing the reference to the popup and implementing `getPopup()` (if necessary) could be pushed up to `AbstractGenericListItemListPanel`.